### PR TITLE
 Create Java code sample for omit email address

### PR DESCRIPTION
**Pull request by sushicw**
Fixes internal bug b/156972853

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [X] Appropriate changes to README are included in PR
- [ ] ~API's need to be enabled to test (tell us)~ *(Nothing new)*
- [ ] ~Environment Variables need to be set (ask us to set them)~ *(Nothing new)*
- [X] Tests pass (`mvn -P lint clean verify`)
  * (Note- `Checkstyle` passing is required; `Spotbugs`, `ErrorProne`, `PMD`, etc. `ERROR`'s are advisory only)
- [X] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Google LLC
```
Comment (gguuss) [inline dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:2]: For new files, use current year, 2020

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import com.google.privacy.dlp.v2.MatchingType;
+import com.google.privacy.dlp.v2.ProjectName;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class InspectStringWithExclusionDict {
+
+  // TODO: Delete me, only for testing
```
Comment (gguuss) [inline dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:44]: Probably should remove before committing this

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      ByteContentItem byteItem =
+          ByteContentItem.newBuilder()
+              .setType(BytesType.TEXT_UTF8)
+              .setData(ByteString.copyFromUtf8(textToInspect))
+              .build();
+      ContentItem item = ContentItem.newBuilder().setByteItem(byteItem).build();
+
+      // Specify the type of info the inspection will look for.
+      List<InfoType> infoTypes = new ArrayList<>();
+      // See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types.
```
Comment (gguuss) [inline dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:78]: I'd replace line 76 with just the reference to the info types

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      for (String typeName : new String[]{"PHONE_NUMBER", "EMAIL_ADDRESS", "CREDIT_CARD_NUMBER"}) {
+        infoTypes.add(InfoType.newBuilder().setName(typeName).build());
+      }
+
+      // Construct an exclusion rule that will exclude matches from the specified excludedMatchList.
```
Comment (gguuss) [inline dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:83]: Drop the repetition of the type, more effective to comment what the code is doing:

```
// Exclude matches from the specified excludedMatchList.
```

**Comment by sushicw** [comment]
Another question I have: should I move the annotation comments to "zoom in" on the part where we construct the params? From the docs I read it seemed like the answer is no, but I worry that this snippet has a ton of boilerplate surrounding the part that we're trying to actually demonstrate. (Note that it's nearly identical to `InspectString.java` except for the additional request params).

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java (hunk 1 above)
Comment (sushicw) [reply dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:2]: Fixed

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java (hunk 2 above)
Comment (sushicw) [reply dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:44]: Fixed

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java (hunk 4 above)
Comment (sushicw) [reply dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:83]: Done

File: dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java (hunk 3 above)
Comment (sushicw) [reply dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java:78]: As clarified offline, moved the comment slightly.

**Comment by gguuss** [comment]
> Another question I have: should I move the annotation comments to "zoom in" on the part where we construct the params? From the docs I read it seemed like the answer is no, but I worry that this snippet has a ton of boilerplate surrounding the part that we're trying to actually demonstrate. (Note that it's nearly identical to `InspectString.java` except for the additional request params).

Yes, I'm in favor of this style of comment

**Pull request merged**